runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineResources.java

```diff
@@ -37,6 +39,11 @@
    * @return A list of absolute paths to the resources the class loader uses.
    */
   public static List<String> detectClassPathResourcesToStage(ClassLoader classLoader) {
+    if (classLoader == ClassLoader.getSystemClassLoader()) {
```

Comment:
this logic https://github.com/apache/geronimo-xbean/blob/trunk/xbean-finder/src/main/java/org/apache/xbean/finder/ClassLoaders.java#L38 can be beneficial. High level idea is to use getResources() to determine what is in the classloader.

Reply:
That isn't guaranteed to work for all jars - `jar M` will create an archive that doesn't contain a `META-INF` folder.

Reply:
This is why it is not the only strategy but likely the most portable i saw accross years. In particular in the context of this method. Do you expect all jars to be in the cp string? Not using manifest to load jars ;)

Reply:
Neither approach is going to work in all cases, but I've had more success with this one.

Reply:
Did you test this class? getResources("") - not only metainf -
 works on most jvm and supports all cases compared to classpath scanning which doesnt support a lot of bootstraps except plain mains :(.

Reply:
I have tested this logic, but I haven't been able to test this specific usage of it. I ran into a similar problem I'm fixing in GoogleCloudPlatform/DataflowJavaSDK#621 and thought it was worth fixing here too.

If you'd like to defer until we're sure it's a problem, or implement the `getResources` approach instead, that's fine with me :)

Reply:
I think this approach seems to solve a legitimate issue and other approaches can be implemented and added.

If you have a good idea on how to test this effectively since its easy to break this kind of functionality without knowing it.

Reply:
to share the idea here what i had in mind high level https://github.com/apache/beam/pull/4514